File: Comms Protocol Java/Comms Protocol Java/comms/protocol/java/Accelerometer_RawDataPacket.java
package comms.protocol.java;

public class Accelerometer_RawDataPacket extends AndroidDataPacket {

	private static int numberOfValues = 3;

	public Accelerometer_RawDataPacket() {
		super(ValidPacketTypes.accelerometer_raw_data_packet, numberOfValues);
	}

	public Accelerometer_RawDataPacket(byte[] payload) throws Exception {
		super(ValidPacketTypes.accelerometer_raw_data_packet, numberOfValues, payload);
	}

	public Accelerometer_RawDataPacket(DataPacket packet) throws Exception {
		super(ValidPacketTypes.accelerometer_raw_data_packet, numberOfValues, packet);
	}
	
	public XYZSensor GetAccelerations() {
		AndroidSensor sensor = DeSerialize();
		return new XYZSensor(sensor.GetValues());
	}
}

File: Comms Protocol CSharp/Comms Protocol CSharp/Accelerometer_RawDataPacket.cs

namespace Comms_Protocol_CSharp
{
    public class Accelerometer_RawDataPacket : AndroidDataPacket
    {
        private static int numberOfVals = 3;

        public Accelerometer_RawDataPacket() : base(ValidPacketTypes.accelerometer_raw_data_packet, numberOfVals)
        {

        }

        public Accelerometer_RawDataPacket(byte[] payload) : base(ValidPacketTypes.accelerometer_raw_data_packet, numberOfVals, payload)
        {

        }

        public Accelerometer_RawDataPacket(DataPacket packet) : base(ValidPacketTypes.accelerometer_raw_data_packet, numberOfVals, packet)
        {

        }

        public XYZSensor GetAcceleration()
        {
            AndroidSensor sensor = DeSerialize();
            return new XYZSensor(sensor.GetValues());
        }
    }
}
